src/test/kotlin/pl/komjago/ticketapp/controllers/BookingControllerTests.kt

```diff
@@ -0,0 +1,432 @@
+        }.andExpect {
+            status { isBadRequest }
+        }
+    }
+
+    @Test
+    fun `make reservation with MakeReservationInput has less than 3 letter name fails name validation`() {
+        val input = MakeReservationInput(
+                listOf(BookedSeatInfo(1, 0)),
+                0,
+                "Ex",
+                "Testerson"
+        )
+
+        mvc.post("/api/booking/book")
+        {
+            contentType = MediaType.APPLICATION_JSON
+            content = mapper.writeValueAsString(input)
+            accept = MediaType.APPLICATION_JSON
+        }.andExpect {
+            status { isBadRequest }
+        }
+    }
+
+    @Test
+    fun `make reservation with MakeReservationInput has name with numbers fails name validation`() {
+        val input = MakeReservationInput(
+                listOf(BookedSeatInfo(1, 0)),
+                0,
+                "Tester1",
+                "Testerson"
+        )
+
+        mvc.post("/api/booking/book")
+        {
+            contentType = MediaType.APPLICATION_JSON
+            content = mapper.writeValueAsString(input)
+            accept = MediaType.APPLICATION_JSON
+        }.andExpect {
+            status { isBadRequest }
+        }
+    }
+
+    @Test
+    fun `make reservation with MakeReservationInput has polish characters in the name passes validation`() {
+        val input = MakeReservationInput(
+                listOf(BookedSeatInfo(1, 0)),
+                0,
+                "Ęęęęęęęęę",
+                "Testerson"
+        )
+
+        val expectedOutput = MakeReservationOutput(
+                toZloty(0.50),
+                LocalDateTime.now()
+        )
+
+        every { service.makeReservation(input) } returns expectedOutput
+
+        mvc.post("/api/booking/book")
+        {
+            contentType = MediaType.APPLICATION_JSON
+            content = mapper.writeValueAsString(input)
+            accept = MediaType.APPLICATION_JSON
+        }.andExpect {
+            status { isCreated }
+        }
+    }
+
+    @Test
+    fun `make reservation with MakeReservationInput has less than 3 letter surname fails surname validation`() {
+        val input = MakeReservationInput(
+                listOf(BookedSeatInfo(1, 0)),
+                0,
+                "Tester",
+                "Ex"
+        )
+
+        mvc.post("/api/booking/book")
+        {
+            contentType = MediaType.APPLICATION_JSON
+            content = mapper.writeValueAsString(input)
+            accept = MediaType.APPLICATION_JSON
+        }.andExpect {
+            status { isBadRequest }
+        }
+    }
+
+    @Test
+    fun `make reservation with MakeReservationInput has surname starting with a small letter fails surname validation`() {
+        val input = MakeReservationInput(
+                listOf(BookedSeatInfo(1, 0)),
+                0,
+                "Tester",
+                "testerson"
+        )
+
+        mvc.post("/api/booking/book")
```

Comment:
czy ścieżka /book na koniec jest potrzebna?

Reply:
Jeśli mowa o endpoincie to nie chiało mi się mapować roota pod posta.